### ISSUE TITLE
metarefresh – "Undefined index: http://macedir.org/entity-category"

### ISSUE THREAD

**Issue by grawity** [simplesamlphp/simplesamlphp#331]
After commit d32eb4a928f5f0e256202ca244cf141411e4489d (I think), metarefresh started logging the following several times per refresh, when trying to load the eduGAIN SP metadata:

```
[f5d0e82bc8] SimpleSAML_Error_Exception: Error 8 - Undefined index: http://macedir.org/entity-category
[f5d0e82bc8] Backtrace:
[f5d0e82bc8] 6 /usr/share/webapps/simplesamlphp/www/_include.php:75 (SimpleSAML_error_handler)
[f5d0e82bc8] 5 /usr/share/webapps/simplesamlphp/lib/SimpleSAML/Utils/Config/Metadata.php:261 (SimpleSAML\Utils\Config\Metadata::isHiddenFromDiscovery)
[f5d0e82bc8] 4 /usr/share/webapps/simplesamlphp/lib/SimpleSAML/Metadata/SAMLParser.php:493 (SimpleSAML_Metadata_SAMLParser::addExtensions)
[f5d0e82bc8] 3 /usr/share/webapps/simplesamlphp/lib/SimpleSAML/Metadata/SAMLParser.php:718 (SimpleSAML_Metadata_SAMLParser::getMetadata20SP)
[f5d0e82bc8] 2 /usr/share/webapps/simplesamlphp/modules/metarefresh/lib/MetaLoader.php:166 (sspmod_metarefresh_MetaLoader::loadSource)
[f5d0e82bc8] 1 /usr/share/webapps/simplesamlphp/modules/metarefresh/www/fetch.php:67 (require)
[f5d0e82bc8] 0 /usr/share/webapps/simplesamlphp/www/module.php:127 (N/A)
```

**Comment by jaimeperez** [comment]
Hi @grawity, thanks for letting us know!

Can you apply this patch and tell me if it solves the issue for you?

```diff
diff --git a/lib/SimpleSAML/Utils/Config/Metadata.php b/lib/SimpleSAML/Utils/Config/Metadata.php
index 3f6b1be..3ac022b 100644
--- a/lib/SimpleSAML/Utils/Config/Metadata.php
+++ b/lib/SimpleSAML/Utils/Config/Metadata.php
@@ -258,7 +258,7 @@ class Metadata
     public static function isHiddenFromDiscovery(array $metadata)
     {
         try {
-            return in_array(self::$HIDE_FROM_DISCOVERY, $metadata['EntityAttributes'][self::$ENTITY_CATEGORY]);
+            return @in_array(self::$HIDE_FROM_DISCOVERY, $metadata['EntityAttributes'][self::$ENTITY_CATEGORY]);
         } catch (\Exception $e) {
         }
         return false;
```

**Comment by grawity** [comment]
Yes, the patch seems to work.

(I also noticed a different error `SimpleSAML_Error_Exception: Error 2 - in_array() expects parameter 2 to be array, null given` with the same backtrace, but the @ silences it as well.)

**Comment by jaimeperez** [comment]
Hi @grawity!

The patch did silence the errors indeed, but made builds fail, given that when making errors silent, exceptions are no longer thrown. I've reverted the code to something hopefully less complex than the original code, but back working with no errors logged. You can find the fix in 69f67dc.
